### PR TITLE
Added iOS Development Roadmap

### LINKED ISSUE #303: iOS Development Roadmap
### Description

Hi, I've added the iOS development roadmap, job boards, communities & open source. I had put all my 1.5 years of experience and knowledge to complete this roadmap.
I'll open PR and link this issue there as well. Please provide me feedback if anything needs to be changed. 

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Hitarthbhatt**
Add:
 -  iOS development roadmap readme.
 - Communities Links.
 - iOS developer job portals.
 - Open source iOS projects.

Closes #303

### CODE REVIEW

File: iOS-Development/Communities/Readme.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+## Communities you can join 👇
+
+- [Community Classroom](https://discord.io/commclassroom)
```
Comment (siddhant-khisty) [inline iOS-Development/Communities/Readme.md:3]: It's `WeMakeDevs` not `Community Classroom`

File: iOS-Development/Job-Boards/Readme.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+## Job Boards you can apply 👇
+
+- [AngleList](https://angel.co/jobs)
```
Comment (siddhant-khisty) [inline iOS-Development/Job-Boards/Readme.md:3]: This looks like a home page for job portals. Please add IOS-specific filters

File: iOS-Development/readme.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+## Networking
+- [Netowking in SwiftUI](https://medium.com/@nutanbhogendrasharma/consume-rest-api-in-swiftui-ios-mobile-app-b3c5d6ecf401)
+
+- [Netowking in UIKit](https://www.freecodecamp.org/news/how-to-make-your-first-api-call-in-swift/)
+
+- [Post API Call](https://www.youtube.com/watch?v=o3Rkg6WmZoY)
+
+- [Advance Networking](https://malcolmkmd.medium.com/writing-network-layer-in-swift-protocol-oriented-approach-4fa40ef1f908)
+
+    #### JSON Coding/Decoding
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:40]: Lint error. Please move it to the start of the line

File: iOS-Development/readme.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+## Database
+- [Core Data Basics](https://medium.com/@ankurvekariya/core-data-crud-with-swift-4-2-for-beginners-40efe4e7d1cc)
+
+- [Core Data](https://youtube.com/playlist?list=PLMRqhzcHGw1aDYKmCuqXQ_IqpWpJlpoJ3)
+
+- [Realm](https://www.kodeco.com/32960966-realm-with-swiftui-tutorial-getting-started)
+
+- [Firebase](https://www.kodeco.com/11609977-getting-started-with-cloud-firestore-and-swiftui)
+    - [Firebase Project](https://youtube.com/playlist?list=PL0dzCUj1L5JEN2aWYFCpqfTBeVHcGZjGw)
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:53]: move it to the start of the line

**Review by siddhant-khisty — CHANGES_REQUESTED** [review]
cc @kaiwalyakoparkar Can you give this a review as well?

File: iOS-Development/Communities/Readme.md (hunk 1 above)
Comment (Hitarthbhatt) [reply iOS-Development/Communities/Readme.md:3]: Sure, It was missed from my end.

File: iOS-Development/readme.md (hunk 3 above)
Comment (Hitarthbhatt) [reply iOS-Development/readme.md:40]: I added these spaces on purpose to look like a sub-point.

File: iOS-Development/readme.md (hunk 4 above)
Comment (Hitarthbhatt) [reply iOS-Development/readme.md]: Fixed

**Comment by kaiwalyakoparkar** [comment]
@siddhant-khisty thank you so much for this review. It should be good if the above are resolved. Please tag me once it's approved by you :)

File: iOS-Development/readme.md (hunk 3 above)
Comment (siddhant-khisty) [reply iOS-Development/readme.md]: @Hitarthbhatt that is not required. the `#`'s used will automatically make it look like a subpoint.

File: iOS-Development/readme.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
+## iOS Development
+
+## Introduction
+- [Intorduction to iOS](https://developer.apple.com/tutorials/app-dev-training)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:5]:
```suggestion
```

File: iOS-Development/readme.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+- [Xcode Setup](https://www.youtube.com/watch?v=vKPCvAPW9Ns)
+
+## Introduction To Swift
+- [Swift Book](https://docs.swift.org/swift-book/LanguageGuide/TheBasics.html)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:10]:
```suggestion
```

File: iOS-Development/readme.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+- [Swift Course](https://www.youtube.com/watch?v=FcsY1YPBwzQ)
+
+## iOS Development Course
+#### SwiftUI (Recommend First) :
+- [Introduction to SwiftUI](https://developer.apple.com/tutorials/swiftui)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:16]:
```suggestion
```

File: iOS-Development/readme.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+- [100 Days Of SwiftUI](https://www.hackingwithswift.com/100/swiftui)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:18]:
```suggestion
```

File: iOS-Development/readme.md [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+- [Hacking with SwiftUI](https://www.hackingwithswift.com/books/ios-swiftui)
+
+#### UIKit :
+- [100 Days Of Swift](https://www.hackingwithswift.com/100)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:23]:
```suggestion
```

File: iOS-Development/readme.md [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+- [Hacking with Swift](https://www.hackingwithswift.com/read)
+
+## View Navigation
+- [SwiftUI Navigation](https://www.kodeco.com/5824937-swiftui-tutorial-navigation)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:28]:
```suggestion
```

File: iOS-Development/readme.md [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+- [UIKit Navigation](https://www.youtube.com/watch?v=LbAd2FIlnos)
+
+## Networking
+- [Netowking in SwiftUI](https://medium.com/@nutanbhogendrasharma/consume-rest-api-in-swiftui-ios-mobile-app-b3c5d6ecf401)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:33]:
```suggestion
```

File: iOS-Development/readme.md [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+- [Netowking in UIKit](https://www.freecodecamp.org/news/how-to-make-your-first-api-call-in-swift/)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:35]:
```suggestion
```

File: iOS-Development/readme.md [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+- [Post API Call](https://www.youtube.com/watch?v=o3Rkg6WmZoY)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:37]:
```suggestion
```

File: iOS-Development/readme.md [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+- [Advance Networking](https://malcolmkmd.medium.com/writing-network-layer-in-swift-protocol-oriented-approach-4fa40ef1f908)
+
+#### JSON Coding/Decoding
+- [JSON Decoder](https://www.avanderlee.com/swift/json-parsing-decoding/)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:42]:
```suggestion
```

File: iOS-Development/readme.md [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+- [JSON Serialization](https://www.hackingwithswift.com/example-code/system/how-to-parse-json-using-jsonserialization)
+
+## Database
+- [Core Data Basics](https://medium.com/@ankurvekariya/core-data-crud-with-swift-4-2-for-beginners-40efe4e7d1cc)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:47]:
```suggestion
```

File: iOS-Development/readme.md [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+- [Core Data](https://youtube.com/playlist?list=PLMRqhzcHGw1aDYKmCuqXQ_IqpWpJlpoJ3)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:49]:
```suggestion
```

File: iOS-Development/readme.md [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+- [Realm](https://www.kodeco.com/32960966-realm-with-swiftui-tutorial-getting-started)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:51]:
```suggestion
```

File: iOS-Development/readme.md [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+- [Firebase](https://www.kodeco.com/11609977-getting-started-with-cloud-firestore-and-swiftui)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:53]:
```suggestion
```

File: iOS-Development/readme.md [hunk 19]
```diff
@@ -0,0 +1,104 @@
...
+- [Firebase Project](https://youtube.com/playlist?list=PL0dzCUj1L5JEN2aWYFCpqfTBeVHcGZjGw)
+
+## Dependancy Manager
+- [Swift Package Manager](https://cocoacasts.com/xcode-fundamentals-how-to-add-a-swift-package-to-a-project)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:58]:
```suggestion
```

File: iOS-Development/readme.md [hunk 20]
```diff
@@ -0,0 +1,104 @@
...
+- [Cocoapods](https://codewithchris.com/cocoapods/)
+
+## Concurrency
+- [Basic's Concepts](https://betterprogramming.pub/concurrency-in-ios-and-swift-guide-50443ce5b0f5)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:63]:
```suggestion
```

File: iOS-Development/readme.md [hunk 21]
```diff
@@ -0,0 +1,104 @@
...
+- [Concurrency in Depth](https://www.freecodecamp.org/news/ios-concurrency/)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:65]:
```suggestion
```

File: iOS-Development/readme.md [hunk 22]
```diff
@@ -0,0 +1,104 @@
...
+- [Async/Await](https://www.avanderlee.com/swift/async-await/)
+
+## Architectures & Design Patterns
+
+- [MVVM Architecture in UIKit](https://www.youtube.com/watch?v=iI0LabCYZJo)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:71]:
```suggestion
```

File: iOS-Development/readme.md [hunk 23]
```diff
@@ -0,0 +1,104 @@
...
+- [MVVM Architecture in SwiftUI](https://azamsharp.medium.com/mvvm-in-swiftui-8a2e9cc2964a)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:73]:
```suggestion
```

File: iOS-Development/readme.md [hunk 24]
```diff
@@ -0,0 +1,104 @@
...
+- [SOLID Principals](https://betterprogramming.pub/swift-s-o-l-i-d-21203ba3a226)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:75]:
```suggestion
```

File: iOS-Development/readme.md [hunk 25]
```diff
@@ -0,0 +1,104 @@
...
+- [Design Patterns](https://aglowiditsolutions.com/blog/top-swift-design-patterns/)
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:77]:
```suggestion
```

File: iOS-Development/readme.md [hunk 26]
```diff
@@ -0,0 +1,104 @@
...
+
+## Most Useful Tools
+- [Xcode Tools & Tricks](https://www.youtube.com/watch?v=ZAqnJQn7xp4): Learn Xocde most essential tools for iOS development.
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:81]:
```suggestion
```

File: iOS-Development/readme.md [hunk 27]
```diff
@@ -0,0 +1,104 @@
...
+- [SourceTree](https://www.sourcetreeapp.com): A Git GUI that offers a visual representation of your repositories. Sourcetree is a free Git client for Windows and Mac.
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:83]:
```suggestion
```

File: iOS-Development/readme.md [hunk 28]
```diff
@@ -0,0 +1,104 @@
...
+- [QuickType.io](https://app.quicktype.io): Convert JSON data into Swift model.
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:85]:
```suggestion
```

File: iOS-Development/readme.md [hunk 29]
```diff
@@ -0,0 +1,104 @@
...
+- [Diawi](https://www.diawi.com): Deploy and test application directly to the devices.
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:87]:
```suggestion
```

File: iOS-Development/readme.md [hunk 30]
```diff
@@ -0,0 +1,104 @@
...
+- [App icon Generator](https://appicon.co): Generate app icons for different devices.
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:89]:
```suggestion
```

File: iOS-Development/readme.md [hunk 31]
```diff
@@ -0,0 +1,104 @@
...
+- [Device Shots](https://deviceshots.com): Design device mockups with the screenshot of your website or mobile application.
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:91]:
```suggestion
```

File: iOS-Development/readme.md [hunk 32]
```diff
@@ -0,0 +1,104 @@
...
+- [Transporter](https://apps.apple.com/us/app/transporter/id1450874784?mt=12): Deplay your iOS applications super fast.
+
+## Practice Projects
+- [Liquid Coder](https://liquidcoder.com/courses): Amazing SwiftUI projects for beginners.
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:96]:
```suggestion
```

File: iOS-Development/readme.md [hunk 33]
```diff
@@ -0,0 +1,104 @@
...
+- [Kevsoft](https://www.youtube.com/c/Kavsoft/videos): Design complex and beautiful UI in SwiftUI.
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:98]:
```suggestion
```

File: iOS-Development/readme.md [hunk 34]
```diff
@@ -0,0 +1,104 @@
...
+- [Spotify Clone](https://youtube.com/playlist?list=PL5PR3UyfTWve9ZC7Yws0x6EGjBO2FGr0o): Complete step by step process to build spotify clone in UIKit.
+
+## Most Useful Links
+- [Apple Developer Videos](https://developer.apple.com/videos/topics/): Why this link is in the last? because it requires basic knowledge of iOS development.
+
```
Comment (siddhant-khisty) [inline iOS-Development/readme.md:103]:
```suggestion
```

**Review by siddhant-khisty — CHANGES_REQUESTED** [review]
Remove all the blank lines between resources. Let there be one empty line between topics.

File: iOS-Development/readme.md (hunk 6 above)
Comment (Hitarthbhatt) [reply iOS-Development/readme.md:10]: Sure, let me fix this quickly.

File: iOS-Development/readme.md (hunk 25 above)
Comment (Hitarthbhatt) [reply iOS-Development/readme.md:56]: Removed

File: iOS-Development/readme.md (hunk 5 above)
Comment (Hitarthbhatt) [reply iOS-Development/readme.md]: Removed

File: iOS-Development/readme.md (hunk 7 above)
Comment (Hitarthbhatt) [reply iOS-Development/readme.md]: Removed

File: iOS-Development/readme.md (hunk 8 above)
Comment (Hitarthbhatt) [reply iOS-Development/readme.md]: Removed

File: iOS-Development/readme.md (hunk 9 above)
Comment (Hitarthbhatt) [reply iOS-Development/readme.md]: Removed

File: iOS-Development/readme.md (hunk 30 above)
Comment (Hitarthbhatt) [reply iOS-Development/readme.md]: Removed

File: iOS-Development/readme.md (hunk 31 above)
Comment (Hitarthbhatt) [reply iOS-Development/readme.md]: Removed

File: iOS-Development/readme.md (hunk 32 above)
Comment (Hitarthbhatt) [reply iOS-Development/readme.md]: Removed

File: iOS-Development/readme.md (hunk 33 above)
Comment (Hitarthbhatt) [reply iOS-Development/readme.md]: Removed

File: iOS-Development/readme.md (hunk 34 above)
Comment (Hitarthbhatt) [reply iOS-Development/readme.md]: Removed

**Review by siddhant-khisty — APPROVED** [review]
LGTM.
@kaiwalyakoparkar please give this a final review and merge

**Pull request merged**